### PR TITLE
Theme save

### PR DESCRIPTION
**Pull request by gagan-aryan**

### CODE REVIEW

**Comment by gagan-aryan** [comment]
@muke1908 what does this checks failed mean?

File: client/src/components/ThemeToggle/index.js [hunk 1]
```diff
@@ -2,11 +2,15 @@ import React, { useContext } from 'react';
 import { ThemeContext } from '../../ThemeContext.js';
 import { FiSun, FiMoon } from 'react-icons/fi';
 import styles from './Style.module.css';
+import storage from '../../utils/storage';
 
 const ThemeToggle = () => {
   const [darkMode, setDarkMode] = useContext(ThemeContext);
 
+  setDarkMode(storage.get('theme'));
```
Comment (DarkSouL11) [inline client/src/components/ThemeToggle/index.js:10]: State should not be set while component renders . You can do it in `useEffect` hook with empty dependencies. It behaves similar to `componentDidMount`

Comment (gagan-aryan) [reply client/src/components/ThemeToggle/index.js:10]: Screenshot from 2020-08-15 15-36-09

I am getting this error now. Not sure what the reason is.

**Comment by muke1908** [comment]
https://github.com/muke1908/chat-e2ee/pull/78#discussion_r470961771
@gagan-aryan I have added a new package. Run `npm install` once.

File: client/src/components/ThemeToggle/index.js [hunk 2]
```diff
@@ -1,12 +1,17 @@
...
 import { FiSun, FiMoon } from 'react-icons/fi';
 import styles from './Style.module.css';
+import storage from '../../utils/storage';
 
 const ThemeToggle = () => {
   const [darkMode, setDarkMode] = useContext(ThemeContext);
 
+  useEffect(() => {
+    setDarkMode(storage.get('theme'));
+  });
```
Comment (DarkSouL11) [inline client/src/components/ThemeToggle/index.js:12]: Need to pass empty dependencies else this effect will be called after every render.

```suggestion
  useEffect(() => {
    setDarkMode(storage.get('theme'));
  }, []);
```

**Comment by DarkSouL11** [comment]
LGTM @muke1908 Can we merge this?

**Comment by muke1908** [comment]
> LGTM @muke1908 Can we merge this?

I was wondering, instead of calling `setDarkMode` in the component, why not while initializing? 
in https://github.com/muke1908/chat-e2ee/blob/master/client/src/ThemeContext.js? 

Looks more appropriate to me.

```
const theme = storage.get('theme');
export const ThemeProvider = (props) => {
  const [darkMode, setDarkMode] = useState(theme);
  return (
    <ThemeContext.Provider value={[darkMode, setDarkMode]}>{props.children}</ThemeContext.Provider>
  );
};
```

**Pull request merged**